lib/commands/location.js

```diff
@@ -13,12 +13,29 @@ export default {
    * if the Location Services are set to 'Always', because the device
    * needs some time to update the location data.
    *
+   * For iOS 17, the return value could be the result of
+   * "mobile:getSimulatedLocation" if the value was set by
+   * "mobile:setSimulatedLocation" already.
+   *
    * @returns {Promise<import('./types').LocationWithAltitude>}
    * @throws {Error} If the device under test returns an error message.
    *                 i.e.: tvOS returns unsupported error
    * @this {XCUITestDriver}
    */
   async getGeoLocation() {
+    // Currently we proxy the setGeoLocation to mobile:setSimulatedLocation for iOS 17+.
+    // It would be helpful to address to use "mobile:getSimulatedLocation" for iOS 17+.
+    if (this.opts.platformVersion && util.compareVersions(this.opts.platformVersion, '>=', '17.0')) {
+      const {latitude, longitude} = await this.mobileGetSimulatedLocation();
```

Comment:
the object deconstruction will throw an exception if the returned value is null

Reply:
The return value itself should have longitude, longitude and location (but always 0) by WDA so `const {latitude, longitude} = null` case does not occur. Error is exception so I think this is ok